cluster-commands.go

```diff
@@ -883,3 +883,98 @@ const (
 	ReplicateRemoveStatusSuccess = "Requested site(s) were removed from cluster replication successfully."
 	ReplicateRemoveStatusPartial = "Some site(s) could not be removed from cluster replication configuration."
 )
+
+type ResyncBucketStatus struct {
+	Bucket    string `json:"bucket"`
+	Status    string `json:"status"`
+	ErrDetail string `json:"errorDetail,omitempty"`
+}
+
+// SRResyncOpStatus - returns status of resync start request.
+type SRResyncOpStatus struct {
+	OpType    string               `json:"op"` // one of "start" or "cancel"
+	ResyncID  string               `json:"id"`
+	Status    string               `json:"status"`
+	Buckets   []ResyncBucketStatus `json:"buckets"`
+	ErrDetail string               `json:"errorDetail,omitempty"`
+}
+
+// SiteResyncOp type of resync operation
+type SiteResyncOp string
+
+const (
+	// SiteResyncStart starts a site resync operation
+	SiteResyncStart SiteResyncOp = "start"
+	// SiteResyncCancel cancels ongoing site resync
+	SiteResyncCancel SiteResyncOp = "cancel"
+)
+
+// SiteReplicationResyncOp - perform a site replication resync operation
+func (adm *AdminClient) SiteReplicationResyncOp(ctx context.Context, site PeerInfo, op SiteResyncOp) (SRResyncOpStatus, error) {
+	reqBytes, err := json.Marshal(site)
+	if err != nil {
+		return SRResyncOpStatus{}, nil
+	}
+
+	v := url.Values{}
+	v.Add("operation", string(op))
+
+	reqData := requestData{
+		relPath:     adminAPIPrefix + "/site-replication/resync/op",
+		content:     reqBytes,
+		queryValues: v,
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodPut, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return SRResyncOpStatus{}, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return SRResyncOpStatus{}, httpRespToErrorResponse(resp)
+	}
+
+	var res SRResyncOpStatus
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	return res, err
+}
+
+// SRResyncStatus - contains resync status info.
+type SRResyncStatus struct {
+	Enabled      bool                                `json:"enabled"`
+	Peer         PeerInfo                            `json:"peer"`
+	ResyncID     string                              `json:"resyncID"`
+	ResyncStatus string                              `json:"resyncStatus"`
+	LastUpdate   time.Time                           `json:"updatedAt"`
+	Stats        map[string]replication.ResyncTarget `json:"stats"`
+}
+
+// SiteReplicationResyncStatus - returns resync status for the deployment id specified as target.
+func (adm *AdminClient) SiteReplicationResyncStatus(ctx context.Context, dID string) (st SRResyncStatus, err error) {
+	v := url.Values{}
+	v.Add("depID", dID)
+
+	reqData := requestData{
+		relPath:     adminAPIPrefix + "/site-replication/resync/status",
+		queryValues: v,
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodGet, reqData)
```

Comment:
As discussed we can move this to `Metrics()` API and provide realtime metrics for this and along with any `mc admin trace --call replication` activity.